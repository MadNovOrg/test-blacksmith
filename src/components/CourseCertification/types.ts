export type ModuleObject = {
  name: string
  completed: boolean
}
