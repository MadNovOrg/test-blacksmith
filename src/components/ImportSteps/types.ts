import { ComponentProps } from 'react'

import { StepsNavigation } from '../StepsNavigation'

export enum ImportStepsEnum {
  CHOOSE = 'choose',
  CONFIGURE = 'configure',
  PREVIEW = 'preview',
  IMPORTING = 'importing',
  RESULTS = 'results',
}
export type Step = {
  label: ComponentProps<typeof StepsNavigation>['steps'][0]['label']
  key: ImportStepsEnum
}
