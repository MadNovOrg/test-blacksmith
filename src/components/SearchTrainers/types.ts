import type { InviteStatus, CourseSchedule } from '@app/types'

export type SearchTrainersSchedule = Partial<
  Pick<CourseSchedule, 'start' | 'end'>
>

export type SearchTrainerBookings = {
  profile_id: string
  status: InviteStatus
  course: {
    schedule: Array<{
      start: string
      end: string
    }>
  }
}
