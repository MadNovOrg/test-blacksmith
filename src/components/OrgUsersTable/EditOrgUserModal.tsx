import {
  Alert,
  Box,
  Button,
  Container,
  FormControlLabel,
  Grid,
  Switch,
  Typography,
} from '@mui/material'
import React, { useCallback, useState } from 'react'
import { useTranslation } from 'react-i18next'

import { useAuth } from '@app/context/auth'
import {
  OrgMembersQuery,
  RemoveOrgMemberMutation,
  RemoveOrgMemberMutationVariables,
  UpdateOrgMemberMutation,
  UpdateOrgMemberMutationVariables,
} from '@app/generated/graphql'
import { useFetcher } from '@app/hooks/use-fetcher'
import { MUTATION as RemoveOrgMemberQuery } from '@app/queries/organization/remove-org-member'
import { MUTATION as UpdateOrgMemberQuery } from '@app/queries/organization/update-org-member'
import theme from '@app/theme'

export type EditOrgUserModalProps = {
  orgMember: OrgMembersQuery['members'][0]
  onClose: () => void
  onChange?: () => void
}

export const EditOrgUserModal: React.FC<
  React.PropsWithChildren<EditOrgUserModalProps>
> = function ({ orgMember, onClose, onChange }) {
  const { t } = useTranslation()
  const { acl } = useAuth()
  const fetcher = useFetcher()
  const [error, setError] = useState<string>()
  const [isAdmin, setIsAdmin] = useState(orgMember.isAdmin)

  const onRemove = useCallback(async () => {
    try {
      await fetcher<RemoveOrgMemberMutation, RemoveOrgMemberMutationVariables>(
        RemoveOrgMemberQuery,
        {
          id: orgMember.id,
        }
      )
      if (onChange) onChange()
      onClose()
    } catch (e: unknown) {
      setError((e as Error).message)
    }
  }, [fetcher, onChange, onClose, orgMember])

  const onSave = useCallback(async () => {
    try {
      await fetcher<UpdateOrgMemberMutation, UpdateOrgMemberMutationVariables>(
        UpdateOrgMemberQuery,
        {
          id: orgMember.id,
          member: { isAdmin: !!isAdmin },
        }
      )
      if (onChange) onChange()
      onClose()
    } catch (e: unknown) {
      setError((e as Error).message)
    }
  }, [fetcher, isAdmin, onChange, onClose, orgMember])

  return (
    <Container>
      {acl.canSetOrgAdminRole() ? (
        <FormControlLabel
          sx={{ py: 2 }}
          control={
            <Switch
              checked={isAdmin ?? false}
              onChange={e => {
                setIsAdmin(e.target.checked)
              }}
              sx={{ px: 2 }}
            />
          }
          label={
            <Box>
              <Typography variant="body1">
                {t(
                  'pages.org-details.tabs.users.edit-user-modal.organization-admin'
                )}
              </Typography>
              <Typography variant="body2" color={theme.palette.grey[700]}>
                {t(
                  'pages.org-details.tabs.users.edit-user-modal.organization-admin-hint'
                )}
              </Typography>
            </Box>
          }
        />
      ) : undefined}

      {error && <Alert severity="error">{error}</Alert>}

      <Grid
        container
        display="flex"
        justifyContent="space-between"
        gap={2}
        mt={2}
      >
        <Button
          data-testid="remove-from-organization"
          type="button"
          variant="outlined"
          color="secondary"
          size="large"
          onClick={onRemove}
        >
          {t(
            'pages.org-details.tabs.users.edit-user-modal.remove-from-organization'
          )}
        </Button>
        <Box>
          <Button
            type="button"
            variant="text"
            color="secondary"
            size="large"
            onClick={onClose}
            sx={{ mr: 2 }}
          >
            {t('common.cancel')}
          </Button>
          <Button
            type="button"
            variant="contained"
            color="primary"
            size="large"
            onClick={onSave}
          >
            {t('common.save-changes')}
          </Button>
        </Box>
      </Grid>
    </Container>
  )
}
