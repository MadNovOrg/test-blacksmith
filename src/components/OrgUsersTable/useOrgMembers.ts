import { gql, useQuery } from 'urql'

import {
  Order_By,
  OrgMembersQuery,
  OrgMembersQueryVariables,
  Organization_Member_Order_By,
} from '@app/generated/graphql'
import { Sorting } from '@app/hooks/useTableSort'
import { SortOrder } from '@app/types'
import { DEFAULT_PAGINATION_LIMIT } from '@app/util'

const QUERY = gql`
  query OrgMembers(
    $offset: Int!
    $limit: Int!
    $orgId: uuid!
    $orderBy: [organization_member_order_by!] = { profile: { createdAt: desc } }
  ) {
    members: organization_member(
      limit: $limit
      offset: $offset
      where: { organization_id: { _eq: $orgId } }
      order_by: $orderBy
    ) {
      id
      profile {
        id
        fullName
        lastActivity
        createdAt
        go1Licenses {
          expireDate
        }
        certificates(where: { status: { _neq: "EXPIRED" } }) {
          id
          courseLevel
          status
          participant {
            certificateChanges {
              payload
            }
          }
        }
      }
      isAdmin
      position
    }

    organization_member_aggregate(where: { organization_id: { _eq: $orgId } }) {
      aggregate {
        count
      }
    }
  }
`

export function useOrgMembers({
  perPage = DEFAULT_PAGINATION_LIMIT,
  offset = 0,
  sort = { dir: 'asc', by: 'createdAt' },
  orgId,
}: {
  perPage?: number
  offset?: number
  orgId: string
  sort?: {
    dir: SortOrder
    by: string
  }
}) {
  const orderBy = getOrderBy(sort)

  const [{ data, fetching }, refetch] = useQuery<
    OrgMembersQuery,
    OrgMembersQueryVariables
  >({
    requestPolicy: 'cache-and-network',
    query: QUERY,
    variables: {
      limit: perPage,
      offset: offset,
      orgId,
      orderBy,
    },
  })

  return {
    members: data?.members,
    fetching,
    total: data?.organization_member_aggregate.aggregate?.count,
    refetch,
  }
}

function getOrderBy({
  by,
  dir,
}: Pick<Sorting, 'by' | 'dir'>): Organization_Member_Order_By {
  const direction = dir === 'asc' ? Order_By.Asc : Order_By.Desc

  switch (by) {
    case 'lastActivity':
      return {
        profile: {
          lastActivity: direction,
        },
      }
    case 'createdAt':
      return { profile: { createdAt: direction } }

    case 'fullName': {
      return { profile: { fullName: direction } }
    }

    default: {
      return { profile: { createdAt: Order_By.Asc } }
    }
  }
}
