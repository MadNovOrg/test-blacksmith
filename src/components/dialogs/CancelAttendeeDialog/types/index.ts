export type FormInput = {
  cancellationFeePercent: number
  cancellationReason: string
}
