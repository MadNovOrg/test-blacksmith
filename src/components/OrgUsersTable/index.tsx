import { CheckCircle } from '@mui/icons-material'
import {
  Box,
  Button,
  Chip,
  CircularProgress,
  Paper,
  Stack,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableRow,
  Typography,
} from '@mui/material'
import Link from '@mui/material/Link'
import React, { useMemo, useState } from 'react'
import { useTranslation } from 'react-i18next'

import { Dialog } from '@app/components/dialogs'
import { EditOrgUserModal } from '@app/components/OrgUsersTable/EditOrgUserModal'
import { ProfileAvatar } from '@app/components/ProfileAvatar'
import { Col, TableHead } from '@app/components/Table/TableHead'
import { TableNoRows } from '@app/components/Table/TableNoRows'
import { useAuth } from '@app/context/auth'
import { OrgMembersQuery } from '@app/generated/graphql'
import { useTablePagination } from '@app/hooks/useTablePagination'
import { useTableSort } from '@app/hooks/useTableSort'
import theme from '@app/theme'
import { CertificateStatus, CourseLevel } from '@app/types'
import { getProfileCertificationLevels } from '@app/util'

import { CertificateStatusChip } from '../CertificateStatusChip'

import { useOrgMembers } from './useOrgMembers'

type OrgUsersTableParams = {
  orgId: string
  onChange?: () => void
  certificateStatus?: CertificateStatus[]
}

export const OrgUsersTable: React.FC<
  React.PropsWithChildren<OrgUsersTableParams>
> = ({ orgId, onChange, certificateStatus }) => {
  const { t } = useTranslation()
  const { acl } = useAuth()

  const sorting = useTableSort('fullName', 'asc')

  const [showEditUserModal, setShowEditUserModal] = useState(false)
  const [editedUser, setEditedUser] = useState<OrgMembersQuery['members'][0]>()

  const { offset, perPage, Pagination } = useTablePagination()

  const { members, fetching, total, refetch } = useOrgMembers({
    perPage,
    offset,
    orgId,
    sort: { by: sorting.by, dir: sorting.dir },
    certificateFilter: certificateStatus,
  })

  const cols = useMemo(() => {
    const _t = (col: string) => t(`pages.org-details.tabs.users.cols-${col}`)
    const mandatoryCols = [
      {
        id: 'fullName',
        label: _t('name'),
        sorting: true,
      },
      {
        id: 'certification',
        label: _t('certification'),
      },
      {
        id: 'go1',
        label: _t('blended-learning'),
      },
      {
        id: 'lastActivity',
        label: _t('last-activity'),
        sorting: true,
      },
      {
        id: 'createdAt',
        label: _t('created-on'),
        sorting: true,
      },
      {
        id: 'permissions',
        label: _t('permissions'),
      },
    ]

    const optionalCols = [
      {
        id: 'actions',
        label: '',
      },
    ]

    return [
      ...mandatoryCols,
      ...(acl.canEditOrgUser([orgId]) ? optionalCols : []),
    ] as Col[]
  }, [acl, t, orgId])

  return (
    <>
      <TableContainer component={Paper} elevation={0}>
        <Table data-testid="organisation-members">
          <TableHead
            cols={cols}
            orderBy={sorting.by}
            order={sorting.dir}
            onRequestSort={sorting.onSort}
          />
          <TableBody>
            {fetching ? (
              <TableRow
                sx={{
                  '&&.MuiTableRow-root': {
                    backgroundColor: 'grey.300',
                  },
                  '&& .MuiTableCell-root': {
                    px: 2,
                    py: 1,
                    color: 'grey.700',
                    fontWeight: '600',
                  },
                }}
              >
                <TableCell colSpan={cols.length}>
                  <Stack direction="row" alignItems="center">
                    <CircularProgress size={20} />
                  </Stack>
                </TableCell>
              </TableRow>
            ) : null}

            <TableNoRows
              noRecords={!fetching && !members?.length}
              itemsName={t('common.users').toLowerCase()}
              colSpan={cols.length}
            />

            {members?.map((member, index) => {
              const certificateLevelsToDisplay = getProfileCertificationLevels(
                member.profile.certificates as {
                  courseLevel: string
                  status: CertificateStatus
                }[]
              )
              const filteredCerts = member.profile.certificates.filter(
                cert =>
                  certificateLevelsToDisplay.indexOf(
                    cert.courseLevel as CourseLevel
                  ) >= 0
              )
              return (
                <TableRow
                  key={member.profile.id}
                  data-testid={`org-member-row-${member.id}`}
                  data-index={index}
                >
                  <TableCell>
                    <Box display="flex" alignItems="center">
                      <ProfileAvatar
                        profile={member.profile}
                        renderLabel={name => (
                          <Box display="flex" flexDirection="column">
                            <Link
                              variant="body2"
                              color={theme.palette.grey[900]}
                              ml={1}
                              href={`/profile/${member.profile.id}?orgId=${orgId}`}
                            >
                              {name}
                            </Link>
                            {member?.position ? (
                              <Typography
                                variant="body2"
                                color="grey.600"
                                ml={1}
                              >
                                {member?.position}
                              </Typography>
                            ) : null}
                          </Box>
                        )}
                      />
                    </Box>
                  </TableCell>
                  <TableCell data-testid="member-certificates">
                    {filteredCerts.map(cert => {
                      const certificationStatus =
                        cert?.status as CertificateStatus
                      const statusTooltip =
                        cert.participant?.certificateChanges[0]?.payload?.note

                      return (
                        <Box
                          key={cert.id}
                          display="flex"
                          alignItems="center"
                          py={1}
                        >
                          <CertificateStatusChip
                            status={certificationStatus}
                            tooltip={statusTooltip}
                          />
                          <Typography variant="body2" ml={1}>
                            {t(
                              `common.certificates.${cert.courseLevel.toLowerCase()}`
                            )}
                          </Typography>
                        </Box>
                      )
                    })}
                  </TableCell>
                  <TableCell>
                    {member.profile.go1Licenses.length === 1 ? (
                      <CheckCircle color="success" />
                    ) : null}
                  </TableCell>
                  <TableCell data-testid="member-last-activity">
                    {t('dates.default', { date: member.profile.lastActivity })}
                  </TableCell>
                  <TableCell data-testid="member-created-at">
                    {t('dates.default', { date: member.profile.createdAt })}
                  </TableCell>
                  <TableCell>
                    <Chip
                      label={
                        member?.isAdmin
                          ? t('pages.org-details.tabs.users.organization-admin')
                          : t('pages.org-details.tabs.users.no-permissions')
                      }
                      color={member?.isAdmin ? 'success' : 'gray'}
                      size="small"
                    />
                  </TableCell>
                  {(member && member.isAdmin) ||
                  (acl.canEditOrgUser([orgId]) &&
                    (acl.canSetOrgAdminRole() || !member.isAdmin)) ? (
                    <TableCell>
                      <Button
                        data-testid="edit-user-button"
                        size="large"
                        variant="text"
                        color="primary"
                        onClick={() => {
                          setEditedUser(member)
                          setShowEditUserModal(true)
                        }}
                      >
                        {t('common.edit')}
                      </Button>
                    </TableCell>
                  ) : null}
                </TableRow>
              )
            })}
          </TableBody>
        </Table>
      </TableContainer>
      {members?.length ? <Pagination total={total ?? 0} /> : null}
      {editedUser ? (
        <Dialog
          maxWidth={800}
          open={showEditUserModal}
          onClose={() => setShowEditUserModal(false)}
          title={t('pages.org-details.tabs.users.edit-user-modal.title', {
            name: editedUser.profile?.fullName,
          })}
          subtitle={
            <Typography color={theme.palette.grey[700]}>
              {t(
                'pages.org-details.tabs.users.edit-user-modal.modification-info'
              )}
            </Typography>
          }
        >
          <EditOrgUserModal
            orgMember={editedUser}
            onClose={() => setShowEditUserModal(false)}
            onChange={async () => {
              refetch()
              if (onChange) onChange()
            }}
          />
        </Dialog>
      ) : null}
    </>
  )
}
