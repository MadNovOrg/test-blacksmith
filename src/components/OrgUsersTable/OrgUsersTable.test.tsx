import React from 'react'
import { MemoryRouter, Route, Routes } from 'react-router-dom'

import { OrgUsersTable } from '@app/components/OrgUsersTable/index'
import { useOrgUsers } from '@app/hooks/useOrgUsers'
import { Organization } from '@app/types'
import { LoadingStatus } from '@app/util'

import { render } from '@test/index'

jest.mock('@app/hooks/useOrgUsers')

const useOrgUsersMock = jest.mocked(useOrgUsers)

describe('OrgUsersTable', () => {
  useOrgUsersMock.mockReturnValue({
    mutate: () => Promise.resolve(undefined),
    status: LoadingStatus.SUCCESS,
    loading: false,
    totalCount: 2,
    error: undefined,
    users: [
      {
        id: '1',
        isAdmin: false,
        createdAt: new Date('2022-04-10T10:00:00').toISOString(),
        updatedAt: new Date('2022-04-10T10:00:00').toISOString(),
        profile: {
          id: '1',
          createdAt: new Date('2022-04-10T10:00:00').toISOString(),
          givenName: 'Given name 1',
          familyName: 'Family name 1',
          fullName: 'Given name 1 Family name 1',
          title: 'Title 1',
          addresses: [],
          attributes: [],
          contactDetails: [],
          email: 'profile1@org.com',
          avatar: 'avatar 1',
          phone: 'phone 1',
          dob: '2000-01-01',
          jobTitle: 'title 1',
          tags: null,
          preferences: [],
          roles: [],
          trainer_role_types: [],
          organizations: [],
          dietaryRestrictions: null,
          disabilities: null,
          lastActivity: new Date('2022-04-10T10:00:00'),
          activeCertificates: {
            aggregate: {
              count: 0,
            },
          },
        },
        organization: {} as Organization,
      },
      {
        id: '2',
        isAdmin: true,
        createdAt: new Date('2022-04-10T10:00:00').toISOString(),
        updatedAt: new Date('2022-04-10T10:00:00').toISOString(),
        profile: {
          id: '2',
          createdAt: new Date('2022-04-10T10:00:00').toISOString(),
          givenName: 'Given name 2',
          familyName: 'Family name 2',
          fullName: 'Given name 2 Family name 2',
          title: 'Title 2',
          addresses: [],
          attributes: [],
          contactDetails: [],
          email: 'profile2@org.com',
          avatar: 'avatar 2',
          phone: 'phone 2',
          dob: '2000-01-01',
          jobTitle: 'title 2',
          tags: null,
          preferences: [],
          roles: [],
          trainer_role_types: [],
          organizations: [],
          dietaryRestrictions: null,
          disabilities: null,
          lastActivity: new Date('2022-04-10T10:00:00'),
          activeCertificates: {
            aggregate: {
              count: 1,
            },
          },
        },
        organization: {} as Organization,
      },
    ],
  })

  it('matches snapshot', async () => {
    const view = render(
      <MemoryRouter initialEntries={['/']}>
        <Routes>
          <Route path="/" element={<OrgUsersTable orgId="1" />} />
        </Routes>
      </MemoryRouter>
    )

    expect(view).toMatchSnapshot()
  })
})
