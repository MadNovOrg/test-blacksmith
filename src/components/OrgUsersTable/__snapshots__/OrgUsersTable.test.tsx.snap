// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrgUsersTable matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiTableContainer-root css-zbd12r-MuiPaper-root-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root css-16un4wc-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-kv444d-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-555rmi-MuiTableRow-root"
            >
              <th
                aria-sort="ascending"
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-s9dfu1-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiButtonBase-root MuiTableSortLabel-root Mui-active css-tf2fk6-MuiButtonBase-root-MuiTableSortLabel-root"
                  role="button"
                  tabindex="0"
                >
                  Name
                  <span
                    class="MuiBox-root css-trfuzi"
                  >
                    sorted ascending
                  </span>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                    data-testid="ArrowDownwardIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-s9dfu1-MuiTableCell-root"
                scope="col"
              >
                Certification
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-s9dfu1-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiButtonBase-root MuiTableSortLabel-root css-tf2fk6-MuiButtonBase-root-MuiTableSortLabel-root"
                  role="button"
                  tabindex="0"
                >
                  Last activity
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                    data-testid="ArrowDownwardIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-s9dfu1-MuiTableCell-root"
                scope="col"
              >
                <span
                  class="MuiButtonBase-root MuiTableSortLabel-root css-tf2fk6-MuiButtonBase-root-MuiTableSortLabel-root"
                  role="button"
                  tabindex="0"
                >
                  Created on
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                    data-testid="ArrowDownwardIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-s9dfu1-MuiTableCell-root"
                scope="col"
              >
                Permissions
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-s9dfu1-MuiTableCell-root"
                scope="col"
              >
                
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-6wwm3k-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-1la21jl-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-11mkwu5-MuiTableCell-root"
                colspan="6"
              >
                <div
                  class="css-1d9cypr-MuiStack-root"
                >
                  <span
                    class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1glked3-MuiCircularProgress-root"
                    role="progressbar"
                    style="width: 20px; height: 20px;"
                  >
                    <svg
                      class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                      viewBox="22 22 44 44"
                    >
                      <circle
                        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                        cx="44"
                        cy="44"
                        fill="none"
                        r="20.2"
                        stroke-width="3.6"
                      />
                    </svg>
                  </span>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiTableContainer-root css-zbd12r-MuiPaper-root-MuiTableContainer-root"
    >
      <table
        class="MuiTable-root css-16un4wc-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-kv444d-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-555rmi-MuiTableRow-root"
          >
            <th
              aria-sort="ascending"
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-s9dfu1-MuiTableCell-root"
              scope="col"
            >
              <span
                class="MuiButtonBase-root MuiTableSortLabel-root Mui-active css-tf2fk6-MuiButtonBase-root-MuiTableSortLabel-root"
                role="button"
                tabindex="0"
              >
                Name
                <span
                  class="MuiBox-root css-trfuzi"
                >
                  sorted ascending
                </span>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                  data-testid="ArrowDownwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-s9dfu1-MuiTableCell-root"
              scope="col"
            >
              Certification
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-s9dfu1-MuiTableCell-root"
              scope="col"
            >
              <span
                class="MuiButtonBase-root MuiTableSortLabel-root css-tf2fk6-MuiButtonBase-root-MuiTableSortLabel-root"
                role="button"
                tabindex="0"
              >
                Last activity
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                  data-testid="ArrowDownwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-s9dfu1-MuiTableCell-root"
              scope="col"
            >
              <span
                class="MuiButtonBase-root MuiTableSortLabel-root css-tf2fk6-MuiButtonBase-root-MuiTableSortLabel-root"
                role="button"
                tabindex="0"
              >
                Created on
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc css-1vweko9-MuiSvgIcon-root-MuiTableSortLabel-icon"
                  data-testid="ArrowDownwardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-s9dfu1-MuiTableCell-root"
              scope="col"
            >
              Permissions
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-s9dfu1-MuiTableCell-root"
              scope="col"
            >
              
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-6wwm3k-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root css-1la21jl-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-11mkwu5-MuiTableCell-root"
              colspan="6"
            >
              <div
                class="css-1d9cypr-MuiStack-root"
              >
                <span
                  class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1glked3-MuiCircularProgress-root"
                  role="progressbar"
                  style="width: 20px; height: 20px;"
                >
                  <svg
                    class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                    viewBox="22 22 44 44"
                  >
                    <circle
                      class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                      cx="44"
                      cy="44"
                      fill="none"
                      r="20.2"
                      stroke-width="3.6"
                    />
                  </svg>
                </span>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "context": Object {
    "auth": Object {
      "activeRole": "user",
      "allowedRoles": Set {
        "user",
      },
      "changeRole": [MockFunction],
      "defaultRole": "user",
      "getJWT": [MockFunction],
      "loadProfile": [MockFunction],
      "loading": false,
      "login": [MockFunction],
      "logout": [MockFunction],
      "profile": Object {
        "addresses": Array [],
        "attributes": Array [],
        "avatar": "",
        "contactDetails": Array [],
        "createdAt": "",
        "dietaryRestrictions": null,
        "disabilities": null,
        "dob": "2002-04-06",
        "email": "john.smith@example.com",
        "familyName": "Smith",
        "fullName": "John Smith",
        "givenName": "John",
        "id": "cacb559d-b85d-5e64-b623-37252520ebda",
        "jobTitle": "Developer",
        "lastActivity": 2020-01-01T00:00:00.000Z,
        "organizations": Array [],
        "phone": "+44 1234555",
        "preferences": Array [],
        "roles": Array [
          Object {
            "role": Object {
              "id": "1",
              "name": "user",
            },
          },
        ],
        "tags": Array [],
        "title": "Mr",
        "updatedAt": "",
      },
      "reloadCurrentProfile": [MockFunction],
    },
  },
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
